src/town_cmd.cpp

```diff
@@ -442,6 +444,8 @@ void ClearAllTownCachedNames()
 static void ChangePopulation(Town *t, int mod)
 {
 	t->cache.population += mod;
+	if (_generating_town) [[unlikely]] return;
```

Comment:
Could this use the existing global `_generating_world` instead of creating a new global?

Reply:
World generation has multiple stages, with town generation near the start. Town population is also updated in the later stages when we 'pre-roll' game ticks for a while, and there we want the signs to be updated.